jena-kafka-connector/src/main/java/org/apache/jena/kafka/common/FusekiProjector.java

```diff
@@ -360,26 +435,43 @@ protected final void commit() {
                                 .append(", ");
             }
             offsetLogMessage.delete(offsetLogMessage.length() - 2, offsetLogMessage.length());
-            long sizeInBytes =
-                    this.eventsSinceLastCommit.stream().map(e -> e.value().sizeInBytes()).reduce(0L, Long::sum);
+            long sizeInBytes = calculateBatchSizeInBytes();
             offsetLogMessage.append(" (")
                             .append(String.format("%,d", this.eventsSinceLastCommit.size()))
                             .append(" events");
             if (sizeInBytes > 0) {
-                Pair<Double, String> sizeDetails = RuntimeInfo.parseMemory(sizeInBytes);
-                offsetLogMessage.append(", ")
-                                .append(String.format("%.2f", sizeDetails.getLeft()))
-                                .append(' ')
-                                .append(sizeDetails.getRight());
+                offsetLogMessage.append(", ").append(FileUtils.byteCountToDisplaySize(sizeInBytes));
             }
             offsetLogMessage.append(")");
             FusekiKafka.LOG.info("{}", offsetLogMessage);
         }
 
+        // Track the sizes of batches to help detect low volume conditions
+        // Only detect this once we have sufficient number of batches observed
+        this.recentBatchSizes.addValue(this.eventsSinceLastCommit.size());
+        if (this.recentBatchSizes.getN() >= recentBatchSizeWindow) {
+            long avgBatchSize = Math.round(this.recentBatchSizes.getMean());
+            if (!this.lowVolumeDetected && avgBatchSize <= lowVolumeBatchSizeThreshold) {
+                FusekiKafka.LOG.info(
+                        "[{}] Average batch size is currently {}, switching to low volume batching mode for these topics",
+                        this.topicNames, avgBatchSize);
+                this.lowVolumeDetected = true;
+            } else if (this.lowVolumeDetected && avgBatchSize > lowVolumeBatchSizeThreshold) {
+                FusekiKafka.LOG.info(
+                        "[{}] Average batch size increased to {}, disabling low volume batching mode for these topics",
+                        this.topicNames, avgBatchSize);
+                this.lowVolumeDetected = false;
+            }
+        }
+
         // Reset our state ready for next batch
         this.eventsSinceLastCommit.clear();
     }
 
+    private Long calculateBatchSizeInBytes() {
```

Comment:
```suggestion
    private long calculateBatchSizeInBytes() {
```

Uber pedantic: We're only ever using the primitive so make the autoboxing take place in the method. It'll never be null.